presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -202,14 +211,36 @@ private static InternalNode chooseLeastBusyNode(List<InternalNode> candidateNode
 
             // choose the preferred node first as long as they're not busy
             if (preferredNodeCount.isPresent() && i < preferredNodeCount.getAsInt() && splitCount < maxSplitCount) {
-                return node;
+                return new NodeResult(Optional.of(node), true);
             }
             // fallback to choosing the least busy nodes
             if (splitCount < min && splitCount < maxSplitCount) {
                 chosenNode = node;
                 min = splitCount;
             }
         }
-        return chosenNode;
+        return new NodeResult(Optional.ofNullable(chosenNode), false);
+    }
+
+    private static class NodeResult
+    {
+        private final Optional<InternalNode> node;
+        private final boolean shouldCache;
```

Comment:
Maybe `cachable`? I understand why `shouldCache` was chosen but `ifShouldCache` seems a bit awkward comparing to `isCachable`. Open to other suggestions as well.